net.longturn.freeciv21.yaml

```diff
@@ -0,0 +1,74 @@
+##########
+# Freeciv21 - net.longturn.freeciv21.yaml
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: James Robertson <jwrober@gmail.com>
+#
+# Build Steps:
+#   - (once) sudo apt install flatpak-builder
+#   - (once) sudo flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+#   - flatpak-builder --install-deps-from=flathub --force-clean build/flatpak dist/net.longturn.freeciv21.yaml
+#   - flatpak-builder --user --install --force-clean build/flatpak dist/net.longturn.freeciv21.yaml
+#   - flatpak run net.longturn.freeciv21
+#   - flatpak remove net.longturn.freeciv21
+#
+##########
+
+app-id: net.longturn.freeciv21
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=home
+  - --filesystem=~/.local/share/freeciv21
+
+command: freeciv21-client
+modules:
+  # Get Lua 5.3 as a dependency not included in the org.kde.Platform package.
+  - name: Lua
+    sources:
+      - type: archive
+        url: https://www.lua.org/ftp/lua-5.3.6.tar.gz
+        sha256: fc5fd69bb8736323f026672b1b7235da613d7177e72558893a0bdcd320466d60
+        dest: include/lua5.3
+    buildsystem: simple
+    build-commands:
+      - cd include/lua5.3 && make linux
+      - cp -r * /app
+
+  # Get Libertunus font as a dependency.
+  - name: Libertinus
+    sources:
+      - type: archive
+        url: https://github.com/alerque/libertinus/releases/download/v7.040/Libertinus-7.040.zip
+        sha256: 2cce08507441d8ae7b835cfe51fb643ad5d9f6b44db4360c4e244f0e474a72f6
+        dest: fonts/Libertinus
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/share/freeciv21/fonts/Libertinus
+      - cp -r fonts/Libertinus/* /app/share/freeciv21/fonts/Libertinus
+
+  # Now build and package freeciv21
+  - name: freeciv21
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_INSTALL_PREFIX=/app
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DFREECIV_DOWNLOAD_FONTS=OFF
+      - -DLUA_INCLUDE_DIR=/app/include/lua5.3/src
+      - -DLUA_LIBRARY=/app/include/lua5.3/src/liblua.a
+    cleanup:
+      - /app/include
```

Comment:
```suggestion
      - /include
```

Reply:
but I don't think it'll have any effect on the lua headers.

Reply:
As you have seen, I copy the headers to `/app` so they persist between module builds. Then I was thinking this would remove it after freeciv21 module is finished as we don't need them any longer once Lua is statically linked to the game.

Reply:
cleanup rules are always "from FLATPAK_DEST" which is in that case `/app` and also only applies to whatever the module installs. Except the top-level one.

Reply:
Yeah, this won't work unless changed as Hubert suggested.